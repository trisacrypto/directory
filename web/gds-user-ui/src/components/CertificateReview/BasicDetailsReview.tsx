import React, { FC, useEffect } from 'react';
import {
  Stack,
  Box,
  Text,
  Heading,
  Table,
  Tbody,
  Tr,
  Td,
  Button,
  Tag,
  Link
} from '@chakra-ui/react';
import { colors } from 'utils/theme';
import { useDispatch, useSelector, RootStateOrAny } from 'react-redux';
import { getStepData } from 'utils/utils';
import { loadDefaultValueFromLocalStorage, TStep } from 'utils/localStorageHelper';
import useCertificateStepper from 'hooks/useCertificateStepper';
import {
  BUSINESS_CATEGORY,
  getBusinessCategiryLabel,
  vaspCategories
} from 'constants/basic-details';
interface BasicDetailsReviewProps {}

const BasicDetailsReview = (props: BasicDetailsReviewProps) => {
  const { jumpToStep } = useCertificateStepper();
  const steps: TStep[] = useSelector((state: RootStateOrAny) => state.stepper.steps);
  const [basicDetail, setBasicDetail] = React.useState<any>({});

  useEffect(() => {
    const getStepperData = loadDefaultValueFromLocalStorage();
    const stepData = {
      website: getStepperData.website,
      established_on: getStepperData.established_on,
      vasp_categories: getStepperData.vasp_categories,
      business_category: getStepperData.business_category
    };

    setBasicDetail(stepData);
  }, [steps]);
  return (
    <Box
      border="1px solid #DFE0EB"
      fontFamily={'Open Sans'}
      bg={'white'}
      color={'#252733'}
      maxHeight={367}
      fontSize={18}
      p={5}
      px={5}>
      <Stack width={'100%'}>
        <Box display={'flex'} justifyContent="space-between" pt={4} ml={5}>
          <Heading fontSize={20} fontFamily="Open Sans">
            Section 1: Basic Details
          </Heading>
          <Button
            bg={colors.system.blue}
            color={'white'}
            onClick={() => jumpToStep(1)}
            height={'34px'}
            _hover={{
              bg: '#10aaed'
            }}>
            Edit
          </Button>
        </Box>
        <Stack fontSize={18}>
          <Table sx={{ 'td:nth-child(2),td:nth-child(3)': { fontWeight: 'bold' } }}>
            <Tbody
              sx={{
                '*': {
                  fontSize: '1rem'
                }
              }}>
              <Tr>
                <Td borderBottom={'none'}>Website</Td>
                <Td borderBottom={'none'}>
                  <Link href={basicDetail.website} isExternal>
                    {basicDetail.website}
                  </Link>
                </Td>
                <Td></Td>
              </Tr>
              <Tr>
                <Td>Business Category</Td>
                <Td>{(BUSINESS_CATEGORY as any)[basicDetail.business_category]}</Td>
                <Td></Td>
              </Tr>
              <Tr borderStyle={'hidden'}>
                <Td>Date of Incorporation/ Establishment</Td>
                <Td>{basicDetail.established_on}</Td>
                <Td></Td>
              </Tr>
              <Tr borderStyle={'hidden'}>
                <Td>VASP Category</Td>
                <Td>
                  {basicDetail?.vasp_categories?.map((categ: any) => {
                    return (
                      <Tag key={categ} color={'white'} bg={'blue.400'} mr={2} mb={1}>
                        {getBusinessCategiryLabel(categ)}
                      </Tag>
                    );
                  })}
                </Td>
                <Td></Td>
              </Tr>
            </Tbody>
          </Table>
        </Stack>
      </Stack>
    </Box>
  );
};
BasicDetailsReview.defaultProps = {
  data: {}
};
export default BasicDetailsReview;
