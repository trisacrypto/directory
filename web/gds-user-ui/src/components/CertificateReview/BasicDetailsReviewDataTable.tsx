import { Stack, Table, Tbody, Tr, Td, Tag, Link } from '@chakra-ui/react';
import { Trans } from '@lingui/react';
import { BUSINESS_CATEGORY, getBusinessCategiryLabel } from 'constants/basic-details';
interface BasicReviewProps {
  data?: any;
}
function BasicDetailsReviewDataTable({ data }: BasicReviewProps) {
  return (
    <Stack fontSize={'1rem'}>
      <Table
        sx={{
          'td:nth-child(2),td:nth-child(3)': { fontWeight: 'semibold' },
          'td:first-child': {
            width: '50%'
          },
          td: {
            borderBottom: 'none',
            paddingInlineStart: 0,
            paddingY: 2.5
          }
        }}>
        <Tbody>
          <Tr>
            <Td pl={'1rem !important'}>
              <Trans id="Organization Name">Organization Name</Trans>
            </Td>
            <Td>{data?.organization_name || 'N/A'}</Td>
            <Td></Td>
          </Tr>
          <Tr>
            <Td borderBottom={'none'} pl={'1rem !important'}>
              <Trans id="Website">Website</Trans>
            </Td>
            <Td borderBottom={'none'} whiteSpace="break-spaces" lineHeight={1.5}>
              {data.website ? (
                <Link href={data.website} isExternal>
                  {data.website}
                </Link>
              ) : (
                'N/A'
              )}
            </Td>
            <Td></Td>
          </Tr>
          <Tr>
            <Td pl={'1rem !important'}>
              <Trans id="Business Category">Business Category</Trans>
            </Td>
            <Td>{(BUSINESS_CATEGORY as any)[data.business_category] || 'N/A'}</Td>
            <Td></Td>
          </Tr>
          <Tr borderStyle={'hidden'}>
            <Td pl={'1rem !important'} whiteSpace="break-spaces" lineHeight={1.5}>
              <Trans id="Date of Incorporation / Establishment">
                Date of Incorporation / Establishment
              </Trans>
            </Td>
            <Td>{data.established_on || 'N/A'}</Td>
            <Td></Td>
          </Tr>
          <Tr borderStyle={'hidden'}>
            <Td pl={'1rem !important'} whiteSpace="break-spaces" lineHeight={1.5}>
              <Trans id="VASP Category">VASP Category</Trans>
            </Td>
            <Td>
              {data?.vasp_categories && data?.vasp_categories.length
                ? data?.vasp_categories?.map((categ: any) => {
                    return (
                      <Tag key={categ} color={'white'} bg={'blue'} mr={2} mb={1}>
                        {getBusinessCategiryLabel(categ)}
                      </Tag>
                    );
                  })
                : 'N/A'}
            </Td>
            <Td></Td>
          </Tr>
        </Tbody>
      </Table>
    </Stack>
  );
}

export default BasicDetailsReviewDataTable;
