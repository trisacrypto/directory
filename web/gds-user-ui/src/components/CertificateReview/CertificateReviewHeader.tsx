import { Box, Heading, Button } from '@chakra-ui/react';
import { Trans } from '@lingui/react';
import useCertificateStepper from 'hooks/useCertificateStepper';

type CertificateReviewHeaderProps = {
  step: number;
  title: string;
};

const CertificateReviewHeader = ({ step, title }: CertificateReviewHeaderProps) => {
  const { jumpToStep } = useCertificateStepper();

  return (
    <Box display={'flex'} justifyContent="space-between" pt={4} ml={0}>
      <Heading fontSize={20} mb="2rem">
        <Trans id={title}>{title}</Trans>
      </Heading>
      <Button
        bg={'blue'}
        color={'white'}
        height={'34px'}
        onClick={() => jumpToStep(step)}
        _hover={{
          bg: '#10aaed'
        }}>
        <Trans id="Edit">Edit</Trans>
      </Button>
    </Box>
  );
};

export default CertificateReviewHeader;
