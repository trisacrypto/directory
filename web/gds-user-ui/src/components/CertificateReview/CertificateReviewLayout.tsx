import { Box, Stack } from '@chakra-ui/react';
import { ReactNode } from 'react';

type CertificateReviewLayoutProps = {
  children: ReactNode;
};

function CertificateReviewLayout({ children }: CertificateReviewLayoutProps) {
  return (
    <Box
      border="1px solid #DFE0EB"
      fontFamily={'Open Sans'}
      color={'#252733'}
      fontSize={18}
      p={5}
      px={5}>
      <Stack>{children}</Stack>
    </Box>
  );
}

export default CertificateReviewLayout;
