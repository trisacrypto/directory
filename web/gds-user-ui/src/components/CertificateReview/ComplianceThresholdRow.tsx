import React from 'react';
import { Text } from '@chakra-ui/react';
import { getFormattedAmount } from 'utils/utils';
import type { ITrixo } from 'modules/dashboard/certificate/entities';

interface ComplianceThresholdProps {
  data:
    | Pick<
        ITrixo,
        'compliance_threshold' | 'compliance_threshold_currency' | 'must_comply_travel_rule'
      >
    | undefined;
}
const ComplianceThresholdRow = (data: ComplianceThresholdProps) => {
  const { compliance_threshold, compliance_threshold_currency, must_comply_travel_rule } =
    data?.data as any;
  const shouldShowComplianceThreshold = compliance_threshold || compliance_threshold !== 0;

  const getAmount = () => {
    return getFormattedAmount(compliance_threshold, compliance_threshold_currency);
  };
  const mustComplyTravelRule = () => {
    if (must_comply_travel_rule && compliance_threshold === 0) {
      return `${getFormattedAmount(compliance_threshold, compliance_threshold_currency)}`;
    }
    return 'N/A';
  };

  return (
    <>
      {shouldShowComplianceThreshold ? (
        <Text>{getAmount()}</Text>
      ) : (
        <Text>{mustComplyTravelRule()}</Text>
      )}
    </>
  );
};

export default ComplianceThresholdRow;
