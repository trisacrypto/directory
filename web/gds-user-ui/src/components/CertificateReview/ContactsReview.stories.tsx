import { Meta, Story } from "@storybook/react";

import ContactsReview from "./ContactsReview";

export default {
  title: "components/ContactsReview",
  component: ContactsReview,
} as Meta;

const Template: Story = (args) => <ContactsReview {...args} />;

export const Default = Template.bind({});
Default.args = {};
