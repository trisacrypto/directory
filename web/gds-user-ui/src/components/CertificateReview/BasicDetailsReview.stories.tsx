import { Meta, Story } from "@storybook/react";
import BasicDetailsReview from "./BasicDetailsReview";

export default {
  title: "components/BasicDetailsReview",
  component: BasicDetailsReview,
} as Meta;

const Template: Story = (args) => <BasicDetailsReview {...args} />;

export const Default = Template.bind({});
Default.args = {};
