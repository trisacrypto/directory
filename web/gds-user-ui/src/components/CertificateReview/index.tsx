/* eslint-disable prefer-reflect */

import React, { useState, useEffect } from 'react';
import { Box, Heading, HStack, Icon, Stack, Text, useToast } from '@chakra-ui/react';

import BasicDetailsReview from './BasicDetailsReview';
import LegalPersonReview from './LegalPersonReview';
import TrisaImplementationReview from './TrisaImplementationReview';
import ContactsReview from './ContactsReview';
import TrixoReview from './TrixoReview';
import FormLayout from 'layouts/FormLayout';
import { RootStateOrAny, useSelector } from 'react-redux';
import useCertificateStepper from 'hooks/useCertificateStepper';
import { hasStepError, mapTrixoFormForBff } from 'utils/utils';
import ReviewSubmit from 'components/ReviewSubmit';
import { registrationRequest } from 'modules/dashboard/certificate/service';
import { loadDefaultValueFromLocalStorage } from 'utils/localStorageHelper';
const CertificateReview = () => {
  const { nextStep, previousStep } = useCertificateStepper();
  const toast = useToast();
  const steps: number = useSelector((state: RootStateOrAny) => state.stepper.steps);
  const hasReachSubmitStep: boolean = useSelector(
    (state: RootStateOrAny) => state.stepper.hasReachSubmitStep
  );
  const [isTestNetSent, setIsTestNetSent] = useState(false);
  const [isMainNetSent, setIsMainNetSent] = useState(false);
  const [result, setResult] = useState('');
  const handleSubmitRegister = async (event: React.FormEvent, network: string) => {
    event.preventDefault();
    try {
      const formValue = loadDefaultValueFromLocalStorage();
      const getMainnetObj = formValue.trisa_endpoint_mainnet;
      const getTestnetObj = formValue.trisa_endpoint_testnet;

      delete formValue.trisa_endpoint_mainnet;
      delete formValue.trisa_endpoint_testnet;
      if (network === 'testnet') {
        formValue.trisa_endpoint = getTestnetObj.endpoint;
        formValue.common_name = getTestnetObj.common_name;
      }
      if (network === 'mainnet') {
        formValue.trisa_endpoint = getMainnetObj.endpoint;
        formValue.common_name = getMainnetObj.common_name;
      }
      // clean form value before send to server
      // const formValueForBff = mapTrixoFormForBff(formValue);
      // console.log('formValueForBff', formValueForBff);
      const response = await registrationRequest(network, mapTrixoFormForBff(formValue));

      if (response.id || response.status === ' "SUBMITTED"') {
        if (network === 'testnet') {
          setIsTestNetSent(true);
          localStorage.setItem('isTestNetSent', 'true');
        }
        if (network === 'mainnet') {
          setIsMainNetSent(true);
          localStorage.setItem('isMainNetSent', 'true');
        }
        setResult(response);

        // toast({
        //   position: 'top-right',
        //   title: 'Success',
        //   description: response.message,
        //   status: 'success',
        //   duration: 5000,
        //   isClosable: true
        // });
      }
    } catch (err: any) {
      console.log('err', err?.response?.data);

      if (!err.response.data.success) {
        toast({
          position: 'top-right',
          title: 'Error Submitting Certificate',
          description: err.response.data.error,
          status: 'error',
          duration: 5000,
          isClosable: true
        });
      } else {
        console.log('something went wrong');
      }
    }
  };

  return (
    <>
      {!hasReachSubmitStep ? (
        <Stack spacing={7}>
          <HStack pt={10}>
            <Heading size="md"> Review </Heading>
            <Box>{/* <Icon as={InfoIcon} color="#F29C36" w={7} h={7} /> (not saved) */}</Box>
          </HStack>
          <FormLayout>
            <Text>
              Please review the information provided, edit as needed, and submit to complete the
              registration form. After the information is reviewed, you will be contacted to verify
              details. Once verified, your TestNet certificate will be issued.
            </Text>
          </FormLayout>
          <BasicDetailsReview />
          <LegalPersonReview />
          <ContactsReview />
          <TrisaImplementationReview />
          <TrixoReview />
        </Stack>
      ) : (
        <ReviewSubmit
          onSubmitHandler={handleSubmitRegister}
          isTestNetSent={isTestNetSent}
          isMainNetSent={isMainNetSent}
          result={result}
        />
      )}
    </>
  );
};

export default CertificateReview;
