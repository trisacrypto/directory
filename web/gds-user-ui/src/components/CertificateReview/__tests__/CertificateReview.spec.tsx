import { render } from 'utils/test-utils';
import CertificateReview from '../index';
describe('<CertificateReview />', () => {
  it('should match snapshot', () => {
    const { container } = render(<CertificateReview />);

    expect(container).toMatchSnapshot();
  });
});
