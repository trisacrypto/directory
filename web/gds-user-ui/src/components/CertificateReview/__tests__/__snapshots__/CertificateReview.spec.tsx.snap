// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CertificateReview /> should match snapshot 1`] = `<div />`;
